tools/module_dependencies.py

```diff
@@ -0,0 +1,127 @@
+__all__ = ['modules_dependent_on', 'dependency_graph']
+
+
+import numpy as np
+import importlib
+import multiprocessing
+import functools
+import sys
+import os
+import glob
+import pytest
+
+
+package = 'skimage'
+included_mods = [
+    # Add private and nested modules here
+    '_shared',
+    'filters.rank',
+]
+excluded_mods = [
+    '__version__',
+]
+
+
+@functools.cache
+def _pkg_modules() -> list[str]:
+    """List all package submodules.
+
+    Returns
+    -------
+    submodules : list
+        Sorted list of package submodules.
+    """
+    pkg = importlib.import_module(package)
+
+    members = {f'{package}.{attr}' for attr in dir(pkg)}
+    included_mods_full = {f'{package}.{mod}' for mod in included_mods}
+    excluded_mods_full = {f'{package}.{mod}' for mod in excluded_mods}
+
+    submodules = sorted((members | included_mods_full) - excluded_mods_full)
+
+    return submodules
+
+
+def _import_dependencies(module: str) -> set[str]:
+    # Check that process is fresh (we've had trouble with it being forked!)
+    bad_modules = {mod for mod in sys.modules if f"{package}." in mod}
+    if bad_modules:
+        msg = f"Expected a fresh process, but {bad_modules} were already imported."
+        raise RuntimeError(msg)
+
+    # Import the module
+    mod = importlib.import_module(module)
+
+    # Import its test modules
+    mod_dir = os.path.dirname(mod.__file__)
+    test_files = glob.glob('**/test_*.py', root_dir=mod_dir, recursive=True)
+    test_modules = [f"{module}/{f}".replace("/", ".")[:-3] for f in test_files]
+    for mod in test_modules:
+        try: importlib.import_module(mod)
+        except pytest.skip.Exception: pass  # raised by `pytest.importorskip`
+
+    # Return the modules that `module` depends on
+    return set(_pkg_modules()) & set(sys.modules)
+
+
+def dependency_graph():
+    """Calculate the dependency graph of the current module.
+
+    Each row represents the dependencies of one subpackage.
```

Comment:
I think this is correct, as long as it is not ambiguous. To avoid any ambiguity:
```suggestion
    Each row represents the dependencies of one subpackage.
    `A[i, j]` is True if module `i` depends on (imports) module `j`.
```
`j` is a dependency of `i`.
`i` is dependent on `j`.